api/logic/evm.go

```diff
@@ -0,0 +1,40 @@
+package logic
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+)
+
+func EVMParseData(data []byte, contractABI *abi.ABI) (*abi.Method, []interface{}, error) {
+	if len(data) < 4 {
+		return nil, nil, fmt.Errorf("data (%dB) too short to have method ID", len(data))
+	}
+	methodID := data[:4]
+	packedArgs := data[4:]
+	method, err := contractABI.MethodById(methodID)
```

Comment:
dang are things convenient once we speak the lingua franca of smart contracts ...